timely_beliefs/beliefs/utils.py

```diff
@@ -749,3 +751,21 @@ def extreme_timedeltas_not_equal(
     if isinstance(td_a, pd.Timedelta):
         td_a = td_a.to_pytimedelta()
     return td_a != td_b
+
+
+def downsample_first(df: pd.DataFrame, resolution: timedelta) -> pd.DataFrame:
+    """Resample data representing instantaneous events.
```

Comment:
This assumption (instantaneous events) is never checked. Or that the index is even a DateTimeIndex

Reply:
Can you reply on this?

Reply:
I'm more leaning towards making this a private function. Would that take away your concern?

Reply:
Not quite. If it matters so much, why not add an `assert` statement? I'm guessing you wouldn't be sure what the outcomes would be if `df` wasn't instantaneous?

Reply:
I opened https://github.com/SeitaBV/timely-beliefs/issues/124 as a follow-up.